lib/streamlit/elements/metric.py

```diff
@@ -0,0 +1,154 @@
+
+        color_and_direction = self.determine_delta_color_and_direction(
+            clean_text(delta_colors), delta
+        )
+        metric_proto.color = color_and_direction.color
+        metric_proto.direction = color_and_direction.direction
+
+        return str(self.dg._enqueue("metric", metric_proto))
+
+    def parse_label(self, label):
+        if not isinstance(label, str):
+            raise TypeError(
+                f"'{str(label)}' is not an accepted type. label only accepts: str"
+            )
+        return label
+
+    def parse_value(self, value):
+        if value is None:
+            return "—"
+        if isinstance(value, float) or isinstance(value, int) or isinstance(value, str):
+            return str(value)
+        else:
+            raise TypeError(
+                f"'{str(value)}' is not an accepted type. value only accepts: "
+                "int, float, str, or None"
+            )
+
+    def parse_delta(self, delta):
+        if delta is None:
+            return ""
+        if isinstance(delta, str):
+            delta = dedent(delta)
+            if delta[0] == "+":
+                return delta[1:]
+            return delta
+        elif isinstance(delta, int) or isinstance(delta, float):
+            return str(delta)
+        else:
+            raise TypeError(
+                f"'{str(delta)}' is not an accepted type. delta only accepts:"
+                " int, float, str, or None"
+            )
+
+    def determine_delta_color_and_direction(self, delta_colors, delta):
+        cd = MetricColorAndDirection(color=None, direction=None)
+
+        if delta is None:
+            cd.color = MetricProto.MetricColor.GRAY
+            cd.direction = MetricProto.MetricDirection.NONE
+            return cd
+
+        if self.is_negative(delta):
+            if delta_colors == "normal":
+                cd.color = MetricProto.MetricColor.RED
+            elif delta_colors == "inverse":
+                cd.color = MetricProto.MetricColor.GREEN
+            elif delta_colors == "off":
+                cd.color = MetricProto.MetricColor.GRAY
+            cd.direction = MetricProto.MetricDirection.DOWN
+        else:
+            if delta_colors == "normal":
+                cd.color = MetricProto.MetricColor.GREEN
+            elif delta_colors == "inverse":
+                cd.color = MetricProto.MetricColor.RED
+            elif delta_colors == "off":
+                cd.color = MetricProto.MetricColor.GRAY
+            cd.direction = MetricProto.MetricDirection.UP
+
+        if cd.color is None or cd.direction is None:
+            raise StreamlitAPIException(
+                f"'{str(delta_colors)}' is not an accepted value. delta_colors only accepts: "
+                "'normal', 'inverse', or 'off'"
+            )
+        return cd
+
+    def is_negative(self, delta):
+        if dedent(str(delta))[0] == "-":
+            return True
+        return False
```

Comment:
This could be more directly written as
```
return dedent(str(delta))[0] == "-"
```
or even better
```
return dedent(str(delta)).startswith("-")
```
which is slightly longer but clearer